util.go

```diff
@@ -0,0 +1,146 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+// DropAll deletes all the data from a database.
+func DropAll(ctx context.Context, conn *sql.Conn, schemaName string) (err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("sqlitedb.DropAll: %w", err)
+		}
+	}()
+
+	if schemaName == "" {
+		schemaName = "main"
+	}
+
+	var indexes, tables, triggers, views []string
+
+	rows, err := conn.QueryContext(ctx, fmt.Sprintf("SELECT name, type FROM %q.sqlite_schema", schemaName))
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var name, sqlType string
+		if err := rows.Scan(&name, &sqlType); err != nil {
+			return err
+		}
+		switch sqlType {
+		case "index":
+			indexes = append(indexes, name)
+		case "table":
+			tables = append(tables, name)
+		case "trigger":
+			triggers = append(triggers, name)
+		case "view":
+			views = append(views, name)
+		default:
+			return fmt.Errorf("unknown sqlite schema type %q for %q", sqlType, name)
+		}
+	}
+	rows.Close()
+	if err := rows.Err(); err != nil {
+		return err
+	}
+
+	for _, name := range indexes {
+		if _, err := conn.ExecContext(ctx, fmt.Sprintf("DROP INDEX %q.%q", schemaName, name)); err != nil {
```

Comment:
I guess %q is close enough?

Reply:
surprisingly close. I believe %q can be used to generate invalid schema and table names, but the escaping appears correct for the set of names it can generate.